src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,63 +2,83 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 import java.math.BigDecimal;
-import java.util.Currency;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 
 import static junit.framework.TestCase.assertEquals;
-import static org.junit.Assert.assertFalse;
+import static org.mockito.Mockito.when;
 
-
-public class ArrayListProductDaoTest
-{
-    private ProductDao productDao;
-    private Product product;
-    private Product productSave;
-    private Currency usd;
+@RunWith(MockitoJUnitRunner.class)
+public class ArrayListProductDaoTest {
+    @Mock
+    private Product product1;
+    @Mock
+    private Product product2;
+    @Mock
+    private Product product3;
+    @Mock
+    private Product product4;
+    @InjectMocks
+    private ArrayListProductDao productDao = ArrayListProductDao.getInstance();
 
     @Before
     public void setup() {
-        productDao = new ArrayListProductDao();
-        usd = Currency.getInstance("USD");
-        product = new Product(3L, "sgs3", "Samsung Galaxy S III", new BigDecimal(300), usd, 5, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S%20III.jpg");
-        productSave = new Product(null, "sgs", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
+        setupProduct(product1, 1L, 1, "Samsung Galaxy S", new BigDecimal(100));
+        setupProduct(product2, 2L, 1, "Apple iPhone", new BigDecimal(200));
+        setupProduct(product3, 3L, 0, "Siemens C56", new BigDecimal(300));
+        setupProduct(product4, null, 1, "Samsung Galaxy S III", new BigDecimal(200));
+        productDao.setProducts(new ArrayList<>(Arrays.asList(product1, product2, product3)));
+    }
+
+    public void setupProduct(Product product, Long id, int stock, String description, BigDecimal lastPrice) {
+        when(product.getId()).thenReturn(id);
+        when(product.getStock()).thenReturn(stock);
+        when(product.getDescription()).thenReturn(description);
+        when(product.getLastPrice()).thenReturn(lastPrice);
     }
 
     @Test
     public void testFindProducts() {
-        List<Product> productList = productDao.findProducts();
-        assertFalse(productList.isEmpty());
+        assertEquals(2, productDao.findProducts().size());
```

Comment:
Please do not test this method by size now. Verify that the expected products are returned in it